### PR TITLE
Using react-query for different API calls + Placeholder on loading 

### PR DESCRIPTION
**Pull request by kalidiagne**
This PR introduce use of `react-query` and also better page loading components when data is loading. 
Also: 
- more optimization and components update
- introduction of `shop` and `leaderboard` page with dummy data

### CODE REVIEW

File: src/hooks/useStore.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { useMutation, useQuery } from "@tanstack/react-query";
+
+export const useFetchStore = () => {
+  return useQuery({
+    queryKey: ["store"],
+    queryFn: async () => {
+      // TODO: replace with actual API call
```
Comment (kalidiagne) [inline src/hooks/useStore.ts:7]: cc: @AndrewCLu @RiverRuby

File: src/hooks/useStore.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      }));
+    },
+  });
+};
+
+export const useRedeemStoreItem = () => {
+  return useMutation({
+    mutationKey: ["redeemStoreItem"],
+    mutationFn: async (itemId: number) => {
+      // TODO: replace with actual API call
```
Comment (kalidiagne) [inline src/hooks/useStore.ts:22]: cc: @AndrewCLu @RiverRuby

**Review by RiverRuby — COMMENTED** [review]
Removing `useTap` and `useAuth` as those functions shouldn't be cached.

**Review by RiverRuby — APPROVED** [review]
This is all amazing!! Cleaned up some small fixes and going to make issues for other things, but overall this is clean AS HELL and it's so nice to see the app come together.

**Pull request merged**